play-json/src/main/scala/play/api/libs/json/JsConstraints.scala

```diff
@@ -173,6 +190,9 @@ trait PathWrites {
   def at[A](path: JsPath)(implicit wrs: Writes[A]): OWrites[A] =
     OWrites[A] { a => JsPath.createObj(path -> wrs.writes(a)) }
 
+  def nonDefault[A](path: JsPath, defaultValue: => A)(implicit wrs: Writes[A]): OWrites[A] =
```

Comment:
I thought about decoupling of configuration. Like 
```
case class JsonConfiguration(
  naming: JsonNaming,
  useDefaultValuesForReads: Boolean = false,
  useDefaultValuesForWrites: Boolean = false)
```
It may expand usage scenarios IMO.